external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -182,28 +170,14 @@ private[kafka010] class KafkaOffsetReader(
         "If startingOffsets contains specific offsets, you must specify all TopicPartitions.\n" +
           "Use -1 for latest, -2 for earliest, if you don't care.\n" +
           s"Specified: ${partitionOffsets.keySet} Assigned: ${partitions.asScala}")
-      logDebug(s"Partitions assigned to consumer: $partitions. Seeking to $partitionOffsets")
+      logDebug(s"Assigned partitions: $partitions. Seeking to $partitionOffsets")
     }
 
     val fnRetrievePartitionOffsets: ju.Set[TopicPartition] => Map[TopicPartition, Long] = { _ =>
       partitionOffsets
     }
 
-    val fnAssertFetchedOffsets: Map[TopicPartition, Long] => Unit = { fetched =>
-      partitionOffsets.foreach {
-        case (tp, off) if off != KafkaOffsetRangeLimit.LATEST &&
-          off != KafkaOffsetRangeLimit.EARLIEST =>
-          if (fetched(tp) != off) {
-            reportDataLoss(
```

Comment:
The `reportDataLoss` is not called in the new version of `def fetchSpecificOffsets`.

Reply:
Please go through reading the doc in the comment on JIRA issue - https://docs.google.com/document/d/1gAh0pKgZUgyqO2Re3sAy-fdYpe_SxpJ6DkeXE8R1P7E/edit?pli=1#

According to his explanation, the check is never be false regardless of the offset.

Reply:
Yeah, I was not able to create a scenario where the condition fired (see the standalone app). If somebody can, feel free to add and we can re-evaluate this code part.

Reply:
As mentioned in the doc already but putting it here as well this condition was never covered w/ testcases. Maybe never worked?! Not sure...